### DISCUSSION TITLE
Authentication strategy and slicer plugin

### DISCUSSION THREAD

**Discussion by tvercaut** [Project-MONAI/MONAILabel#102] [Ideas]
Given the client/server nature of MONAILabel, it would be interesting to implement some functionality for user authentication. Is this on the roadmap?

In any case, some documentation about its status in MONAILabel would be very useful.

I am particularly interested in the slicer plugin but smilar questions would arise with other client integrations (e.g. OHIF).

**Comment by SachidanandAlle** [comment]
MONAILabel provides simple endpoints for users to hook in and run simple AI related tasks.  Otherwise currently it doesn't fall under the scope to add authentication (user, role management etc) as part of this project.  However there are multiple frameworks/solutions under apache/nginx to support this feature.

For e.g. https://www.keycloak.org/ I do believe authentication itself is a seperate/broader level project and it doesn't fall under MONAILabel.

**Comment by tvercaut** [comment]
Thanks. That's fine. It would be helpful to document it explicitly. Maybe have a scope section in the readme with an explicit list of what is considered out of scope? Maybe also an FAQ for further details (e.g. pointers to keycloak, etc.)?

**Comment by dean-taylor** [comment]
Currently MONAI Label supports OAuth2 Resource Owner Credentials Grant flow with simple authorisation to separate a number of roles for management and usage. I have added a more refined discussion to extend this functionality to include the current industry best practice and standard. Specifically OAuth2 Authorisation Code Grant with PKCE.
